testutil/daemon/daemon_unix.go

```diff
@@ -13,17 +13,17 @@ import (
 	"gotest.tools/assert"
 )
 
-func cleanupNetworkNamespace(t testing.TB, execRoot string) {
+func cleanupNetworkNamespace(t testing.TB, d *Daemon) {
 	t.Helper()
 	// Cleanup network namespaces in the exec root of this
 	// daemon because this exec root is specific to this
 	// daemon instance and has no chance of getting
 	// cleaned up when a new daemon is instantiated with a
 	// new exec root.
-	netnsPath := filepath.Join(execRoot, "netns")
+	netnsPath := filepath.Join(d.execRoot, "netns")
 	filepath.Walk(netnsPath, func(path string, info os.FileInfo, err error) error {
 		if err := unix.Unmount(path, unix.MNT_DETACH); err != nil && err != unix.EINVAL && err != unix.ENOENT {
```

Comment:
While at it, we can switch to using `mount.Unmount()` which does all the same things.

Yet better, use `RecursiveUnmount()` to further simplify things here.

Reply:
I now recall I had a PR that made changes here; https://github.com/moby/moby/pull/36511, but for some reason it kept failing.

Let me try using the RecursiveUnmount() that sounds like a clean approach (if it works)

Reply:
I take that back. Neither `filepath.Walk()` nor `RecursiveUnmount()` should be used here, as per https://github.com/moby/moby/pull/36511#discussion_r172975216

Reply:
I.e. the approach in #36511 is the correct one; I would take that one here )

Reply:
Let me rebase the other one, to keep this one simpler (and in case it's still problematic). I can rebase it on top of this one